server/apps/pa_img/pa_img_datalayer.py

```diff
@@ -76,13 +96,18 @@ def find(self, resource, req, lookup):
         :return:
         """
 
-        url = self._app.config['PAIMG_SEARCH_URL'] + '/search'
+        url = self._app.config['PAIMG_SEARCH_URL']
         fields = {}
         if 'query' in req['query']['filtered']:
-            query_keywords = req['query']['filtered']['query']['query_string']['query']
-            query_keywords = query_keywords.replace('slugline:', 'keywords:')
-            query_keywords = query_keywords.replace('description:', 'caption:')
-            fields = extract_params(query_keywords, ['headline', 'caption', 'keywords'])
+            query = req['query']['filtered']['query']['query_string']['query'] \
+                .replace('slugline:', 'keywords:') \
+                .replace('description:', 'caption:')
+            for key in ('headline', 'keywords', 'caption'):
```

Comment:
If you search in all text fields by 'test', it will appear at the top of the list x('test')
If then you search by headline 'collins' the x('test') is still at the top of the list but actually it is not used.